File: language/src/main/java/brj/nodes/SetNode.java
package brj.nodes;

import brj.BridjeLanguage;
import brj.runtime.BridjeSet;
import com.oracle.truffle.api.dsl.Specialization;

public abstract class SetNode extends CollNode {

    protected SetNode(BridjeLanguage lang) {
        super(lang);
    }

    @Override
    @Specialization
    BridjeSet executeColl(Object[] res) {
        return new BridjeSet(res);
    }
}

File: language/src/main/kotlin/brj/nodes/SetNode.kt
package brj.nodes

import brj.BridjeLanguage
import brj.runtime.BridjeSet
import com.oracle.truffle.api.dsl.Specialization
import com.oracle.truffle.api.source.SourceSection

abstract class SetNode(lang: BridjeLanguage, loc: SourceSection?) : CollNode(lang, loc) {
    @Specialization
    override fun executeColl(els: Array<Any>) = BridjeSet(els)
}